test/new-e2e/tests/agent-platform/common/utils.go

```diff
@@ -0,0 +1,170 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	e2eClient "github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	pkgmanager "github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common/pkg-manager"
+	svcmanager "github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common/svc-manager"
+	"gopkg.in/yaml.v2"
+)
+
+// ServiceManager generic interface
+type ServiceManager interface {
+	Status(service string) (string, error)
+	Start(service string) (string, error)
+	Stop(service string) (string, error)
+	Restart(service string) (string, error)
+}
+
+// PackageManager generic interface
+type PackageManager interface {
+	Remove(pkg string) (string, error)
+}
+
+// FileManager generic interface
+type FileManager interface {
+	ReadFile(path string) (string, error)
+	CheckFile(path string) (string, error)
+	FindFileInFolder(path string) (string, error)
+	WriteFile(path string, content string) (string, error)
+}
+
+// Helper generic interface
+type Helper interface {
+	GetInstallFolder() string
+	GetConfigFolder() string
+	GetBinaryPath() string
+}
+
+func getServiceManager(vmClient *e2eClient.VMClient) ServiceManager {
+	if _, err := vmClient.ExecuteWithError("systemctl --version"); err == nil {
+		return svcmanager.NewSystemctlSvcManager(vmClient)
+	}
+	return nil
+}
+
+func getPackageManager(vmClient *e2eClient.VMClient) PackageManager {
+	if _, err := vmClient.ExecuteWithError("apt --version"); err == nil {
+		return pkgmanager.NewAptPackageManager(vmClient)
+	}
+	return nil
+}
+
+// ExtendedClient contain the Agent Env and SvcManager and PkgManager for tests
+type ExtendedClient struct {
+	VMClient    *e2eClient.VMClient
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Wondering if the VM should be embedded. We could then use its testing context

Reply:
`VMClient` is fine, I was wondering if you should embed it rather than have it as a member

Reply:
Oh I see, maybe we could. I am not sure to understand exactly what would be the advantage of reusing the VMClient testing context. We can still modify it in another PR!